server/http.go

```diff
@@ -480,21 +476,34 @@ func (p *Plugin) postAuthenticationMessage(channelID string, userID string, mess
 	return p.API.SendEphemeralPost(userID, post)
 }
 
-func (p *Plugin) checkPreviousMessages(channelID string) (recentMeeting bool, meetindID int, creatorName string, err *model.AppError) {
+func (p *Plugin) checkPreviousMessages(channelID string) (recentMeeting bool, meetingLink string, creatorName string, provider string, err *model.AppError) {
 	var zoomMeetingTimeWindow int64 = 30 // 30 seconds
 
 	postList, appErr := p.API.GetPostsSince(channelID, (time.Now().Unix()-zoomMeetingTimeWindow)*1000)
 	if appErr != nil {
-		return false, 0, "", appErr
+		return false, "", "", "", appErr
 	}
 
 	for _, post := range postList.ToSlice() {
-		if meetingID, ok := post.Props["meeting_id"]; ok {
-			return true, int(meetingID.(float64)), post.Props["meeting_creator_username"].(string), nil
+		if meetingProvider, ok := post.Props["meeting_provider"]; ok {
+			if meetingProvider == zoomProviderName {
+				if meetingLink, ok := post.Props["meeting_link"]; ok {
+					return true, meetingLink.(string), post.Props["meeting_creator_username"].(string), meetingProvider.(string), nil
```

Comment:
Should we be checking `meeting_creator_username` before casting?

Reply:
The assumption here is that a Zoom meeting will always have a meeting creator. We can put a check here to be extra careful if you want.

Reply:
+1 for checking before using the type-asserted value